commands/password-managers/bitwarden/copy-first-matching-totp.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+# Dependencies:
+#   1. The Bitwarden CLI: https://bitwarden.com/help/article/cli/
+#   2. The `jq` utility: https://stedolan.github.io/jq/
+#
+# Install via homebrew: `brew install bitwarden-cli jq`
+
+# Required parameters:
+# @raycast.schemaVersion 1
+# @raycast.title Copy First Matching TOTP
+# @raycast.mode silent
+#
+# Optional parameters:
+# @raycast.packageName Bitwarden
+# @raycast.icon images/bitwarden.png
+# @raycast.argument1 { "type": "text", "placeholder": "Query" }
+#
+# Documentation
+# @raycast.author Adrian Reyes
+# @raycast.authorURL https://github.com/int3rrupt
+# @raycast.description Search all items in a Bitwarden vault, and copy the TOTP of the first search result to the clipboard.
+
+itemNotFound() {
+  echo "The query '${BASH_ARGV[0]}' did not return any items."
+  exit 1
+}
+
+itemNoTotp() {
+  echo "Item '$1' does not have a TOTP configured."
+  exit 1
+}
+
+if ! command -v bw &> /dev/null; then
+  echo "The Bitwarden CLI is not installed."
+  exit 1
+elif ! command -v jq &> /dev/null; then
+  echo "The jq utility is not installed."
+  exit 1
+fi
+
+token=$(security find-generic-password -a ${USER} -s raycast-bitwarden -w 2> /dev/null)
+token_status=$?
+
+session=""
+if [ $token_status -eq 0 ]; then
+  session="--session $token"
+fi
+
+bw unlock --check $session > /dev/null 2>&1
+unlocked_status=$?
+
+if [ $unlocked_status -ne 0 ]; then
+  echo "Vault is locked!"
+  exit 1
+fi
+
+item=$(bw list items --search "$1" $session 2> /dev/null | jq ".[0] | { id: .id, name: .name }")
+name=$(echo $item | jq --exit-status ".name") || itemNotFound
+id=$(echo $item | jq --exit-status ".id") || itemNotFound
+totp=$(eval "bw get totp $id $session" 2> /dev/null) || itemNoTotp "$name"
```

Comment:
There's absolutely nothing functionally wrong with this, as far as I can tell, but the `bw list items` command will return the TOTP in it's result. If we copy it from there instead of the `id`, then we can make one less call to the CLI.
```suggestion
item=$(bw list items --search "$1" $session 2> /dev/null | jq ".[0] | { name: .name, totp: .login.totp }")
name=$(echo $item | jq --exit-status ".name") || notFound
totp=$(echo $item | jq --raw-output --exit-status ".totp") || notFound
```

Reply:
The actual TOTP isn't provided in the JSON which is why we need `bw get totp`. What you get in the JSON is the TOTP key/secret that is used to generate the TOTP. I also don't see any option to include it in the JSON output. Have you seen it in the JSON? Am I missing something?

Reply:
Ah, yes that makes more sense than assuming the `bw list items` JSON output would change every 30 seconds 😂 . I haven't seen any other TOTP-related fields in the JSON, but it doesn't matter anyway because what you have should work just fine.